### PR TITLE
made index field evaluatable

### PR DESCRIPTION
**Pull request by danrot**
__tasks:__

- [x] test coverage (index field in search results)
- [x] gather feedback for my changes
- [x] submit changes to the documentation
- [x] add index name to search results in elastic search (check if obtainable from elastic result, or if it has to be saved as a field as in Zend Lucene)

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | none
| BC breaks        | index tag changed in xml mapping
| Documentation PR | none

### CODE REVIEW

**Comment by wachterjohannes** [comment]
@danrot this don`t work for snippets (maybe phpcr does it) and medias ...

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
 sudo: false
 
 php:
-  - 5.4
+  - 5.5
```
Comment (wachterjohannes) [inline .travis.yml]: shouldnt we add here also 5.6?

File: Resources/docs/introduction.rst [hunk 2]
```diff
@@ -103,7 +103,7 @@ Place the following mapping file in the
     <massive-search-mapping xmlns="http://massive.io/schema/dic/massive-search-mapping">
 
         <mapping class="Model\Product">
-            <index name="product" />
+            <index value="product" />
```
Comment (wachterjohannes) [inline Resources/docs/introduction.rst]: why is this here now value - i think "name" would be a better name for this property!

File: Search/Document.php [hunk 3]
```diff
@@ -260,7 +260,6 @@ public function jsonSerialize()
             'url' => $this->url,
             'image_url' => $this->imageUrl,
             'locale' => $this->locale,
-            'category' => $this->category,
```
Comment (wachterjohannes) [inline Search/Document.php]: you have replaced the category with index - didnt you need the index here?

File: Search/LocalizationStrategyInterface.php [hunk 4]
```diff
@@ -20,18 +20,31 @@
      * Provide localized version of the given index name.
      *
      * @param string $indexName
+     * @param string $locale
+     *
+     * @return
```
Comment (wachterjohannes) [inline Search/LocalizationStrategyInterface.php]: empty return type

File: Search/LocalizationStrategyInterface.php [hunk 5]
```diff
@@ -20,18 +20,31 @@
...
      */
     public function localizeIndexName($indexName, $locale);
 
     /**
+     * Remove the localization part of the given index name.
+     *
+     * @param $indexName
+     *
+     * @return mixed
```
Comment (wachterjohannes) [inline Search/LocalizationStrategyInterface.php]: i think this should be string?

File: .travis.yml (hunk 1 above)
Comment (danrot) [inline .travis.yml]: That's probably a good idea, will do it.

File: Resources/docs/introduction.rst (hunk 2 above)
Comment (danrot) [inline Resources/docs/introduction.rst]: Because the `FieldEvaluator` evaluates the property differently based on the attribute name. I only adjusted the index tag to the other ones, where name is handled differently than on the index tag.

File: Search/Document.php (hunk 3 above)
Comment (danrot) [inline Search/Document.php]: No, because the search document already knows from which index it is delivered.
